package.json

```diff
@@ -39,9 +39,10 @@
   "scripts": {
     "bump": "beachball bump",
     "change": "beachball change",
-    "check": "beachball check",
+    "checkchange": "beachball check  --scope \"!packages/*\" --changehint \"Run 'yarn change' to generate a change file\"",
+    "check": "beachball check ",
     "publish": "beachball publish",
-    "publish-ci": "beachball publish -n $(npm.token)",
+    "publish-ci": "beachball publish -b origin/master",
```

Comment:
Does beachball actually make commits to master like Lerna does?

Reply:
I have not used Learna before. After you fill out the scope for file changed, it will automatically commit. But always check with git status to make sure.

Reply:
@nicholasrice this link presents the workflow in detail: https://microsoft.github.io/beachball/overview/getting-started.html

Reply:
@chrisdholt So yes? What I'm trying to understand is if there is a race condition between someone manually merging a PR prior and beachball's publishing commit.

Reply:
I think there could be, though it seems like that's acknowledged and they try to account for that discrepancy (sorry for another link): https://microsoft.github.io/beachball/cli/publish.html

> In most version bumping systems that automates git repo version with npm registry, they assume that the source code is fresh by the time they push the changes back to the git repository. This is almost never the case when the build takes more than a few minutes! So, beachball fetches latest before we push back to the target branch. In large systems, it has been observed that without the git fetch, it becomes a source of conflict.

FWIW (which may be very little), we've been using this on the Fluent UI side with both automated nightly publishes as well as manual publishes during the day. We haven't hit an issue with those *yet*...but I acknowledge, *yet*.

Reply:
Gotcha okay, well hopefully in practice it won't be an issue.

Reply:
@nicholasrice totally agree :). I do think there are a couple things which limit the risk (though there isn't inherently *no risk*):

1. Only contributors with write access have merge permissions. We can manage through policy in terms of *manual* publishes...if we we need to manually publish we ask folks to hold PR's.

2. We setup the nightly publish at a time which is *inconvenient* for merging anyways. Perhaps something like 2AM PST (5AM ET), etc...

Reply:
For 2, I think folks who *can merge* would be aware of this and ideally try and avoid merging at that time

Reply:
Lastly (I hope), If we see a manual publish has already been kicked off - we don't run it again while it's already going. 

I think those simple guidelines would help avoid the issue